### ISSUE TITLE
App crashing when extracting the zip 

### ISSUE THREAD

**Issue by stellasoft-will** [driftyco/ionic-plugin-deploy#45]
I/IONIC.DEPLOY.EXTRACT(16376): Extracting update
I/IONIC.DEPLOY.UNZIP(16376): 1d172e0a893211e59ceb82f519894c00
I/IONIC.DEPLOY.HASVER(16376): Checking 1d172e0a893211e59ceb82f519894c00...
I/IONIC.DEPLOY.HASVER(16376): 5988b63087b911e5adae6ea7746e6e01 == 1d172e0a893211e59ceb82f519894c00
I/IONIC.DEPLOY.HASVER(16376): 6536a4f0892c11e5bec9364d093b8832 == 1d172e0a893211e59ceb82f519894c00
I/IONIC.DEPLOY.HASVER(16376): No
I/IONIC.DEPLOY.UNZIP_DIR(16376): /data/data/com.herdkeeper.sfh/app_1d172e0a893211e59ceb82f519894c00
I/IONIC.DEPLOY.ENTRIES(16376): Total: 923
I/IONIC.DEPLOY.EXTRACT(16376): Progress: 0%
E/Watchdog(  763): !@Sync 126
I/IONIC.DEPLOY.UNZIP_STEP(16376): Exception: /data/data/com.herdkeeper.sfh/app_1d172e0a893211e59ceb82f519894c00/css/dhtmlxscheduler.css: open failed: ENOTDIR (Not a directory)
I/chromium(16376): [INFO:CONSOLE(173)] "Ionic Deploy: Progress... ", source: file:///android_asset/www/plugins/cordova-plugin-console/www/console-via-logger.js (173)
D/PluginManager(16376): exec() call to unknown plugin: Console
I/IONIC.DEPLOY.REMOVE(16376): Removed www.zip
I/IONIC.DEPLOY.REDIRECT(16376): Loading deploy version: 1d172e0a893211e59ceb82f519894c00
E/cutils  (  226): Failed to mkdirat(/storage/extSdCard/Android): Read-only file system
W/ContextImpl(16376): Failed to ensure directory: /storage/extSdCard/Android/data/com.herdkeeper.sfh/files
E/cutils  (  226): Failed to mkdirat(/storage/extSdCard/Android): Read-only file system
W/ContextImpl(16376): Failed to ensure directory: /storage/extSdCard/Android/data/com.herdkeeper.sfh/cache
W/Vold    (  226): Returning OperationFailed - no handler for errno 30
W/Vold    (  226): Returning OperationFailed - no handler for errno 30

this is the log cat error

**Comment by stellasoft-will** [comment]
looks like ths issue is with extracting the css file, ps subbed in com.example.test, also running on ionic cli alpha 2.0.0-alpha.18

**Comment by nikis** [comment]
Today i had the same problem using ionic-plugin-deploy 0.5.0, ionic cli version 1.7.14 and custom zip archive (not created with ionic upload command line). The easies way to fix is to find your IonicDeploy.java and check if ZipEntry is directory and if is directory the loop needs to continue.  https://gist.github.com/nikis/01351e93f0551ba3ff5ed10d62938e27 Line 629:633

**Comment by netsesame2** [comment]
Yes, it is a clear mistake in the code. So strange is it exist there for a long time.

**Comment by piotrowski** [comment]
There was a change at this code a few days ago:

https://github.com/driftyco/ionic-plugin-deploy/commit/03330af1f5a080601fd6e324f796732fec2058d6#diff-989d837c7890332b380d46fd2fa0f063

Now there is a check if the size is != 0 wrapped around the extraction logic. Does this also cover the issue you posted about last year? Or should your change still be added?

Note: The change would be adding 
```
        if(zipEntry.isDirectory()) {
        	extracted += 1;
        	zipInputStream.closeEntry();
        	continue;
        }
```
before this line: `byte[] buffer = new byte[2048];`
